### PR TITLE
MEN-4514: Include session logging in remote terminal documentation

### PR DESCRIPTION
**Pull request by alfrunes**
changelog: title

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

### CODE REVIEW

File: 09.Add-ons/01.Remote-Terminal/docs.md [hunk 1]
```diff
@@ -76,6 +76,13 @@ is to read messages from the websocket, route them by session id
 to the correct shell, read the output going to the terminal,
 and pass it upstream over the websocket.
 
+For customers with Mender Enterprise, audit logs are collected for all terminal
```
Comment (merlin-northern) [inline 09.Add-ons/01.Remote-Terminal/docs.md:79]: suggested change: "For customers with Mender Enterprise" -> "For _Mender Enterprise customers_"

File: 09.Add-ons/01.Remote-Terminal/docs.md [hunk 2]
```diff
@@ -76,6 +76,13 @@ is to read messages from the websocket, route them by session id
 to the correct shell, read the output going to the terminal,
 and pass it upstream over the websocket.
 
+For customers with Mender Enterprise, audit logs are collected for all terminal 
+sessions. The audit logs include terminal creation/termination events as well as 
+the raw terminal output, for which you can replay in the browser from the Mender UI.
```
Comment (merlin-northern) [inline 09.Add-ons/01.Remote-Terminal/docs.md:81]: 1. saying "raw terminal output" will raise questions, I would drop "raw". Also technically speaking it is not the auditlogs that contain the session recording as it is. 
2. suggested change: "terminal output, for which you can"  -> "terminal output which you can"

File: 09.Add-ons/01.Remote-Terminal/40.Security/docs.md [hunk 3]
```diff
@@ -50,6 +50,6 @@ Remote Terminal does not open nor listen on any port: the device initiates the c
 !!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
 !!! for an overview of all Mender plans and features.
 
-Audit logs hold a list of every start and stop of a session. You can access this information
-in the Mender UI.
-
+Mender audits every remote terminal connection by logging the start/stop events
+as well as the terminal output. The audit logs are available in the Mender UI "AUDIT LOG" tab.
+There you can replay the recorded terminal session in real time.
```
Comment (merlin-northern) [inline 09.Add-ons/01.Remote-Terminal/40.Security/docs.md:55]: five comments here:

1. I would start with "Every Remote Terminal invocation is saved in the audit logs"
2. Perhaps we shall create a link referring  to the audit logs for further reading?
3. every time I say "replay" people ask: "ok, so those commands are going to be executed on the device again?" I have a strong feeling that we have to mention the fact that it is not the case.
4. As far as I know we also provide the "download" of the recording, maybe it is worthwhile to mention that?
5. I would mention that the same holds for mender-cli

File: 09.Add-ons/01.Remote-Terminal/docs.md [hunk 4]
```diff
@@ -76,6 +76,13 @@ is to read messages from the websocket, route them by session id
...
+
+!!! Audit logs are available to customers with
```
Comment (merlin-northern) [inline 09.Add-ons/01.Remote-Terminal/docs.md:83]: minor:
```suggestion
!!! Audit logs are only available in the Mender Enterprise plan.
```
(reasoning: to be consistent with what we write in RBAC "`Role Based Access Control is only available in the Mender Enterprise plan.`")

File: 09.Add-ons/01.Remote-Terminal/40.Security/docs.md (hunk 3 above)
Comment (alfrunes) [reply 09.Add-ons/01.Remote-Terminal/40.Security/docs.md:55]: > 1.   I would start with "Every Remote Terminal invocation is saved in the audit logs"
> 2.    Perhaps we shall create a link referring to the audit logs for further reading?
> 3.   every time I say "replay" people ask: "ok, so those commands are going to be executed on the device again?" I have a strong feeling that we have to mention the fact that it is not the case.

:+1:

> 4.    As far as I know we also provide the "download" of the recording, maybe it is worthwhile to mention that?
> 5.    I would mention that the same holds for mender-cli

I don't know how to accurately document this. The record you download from the UI is incompatible with mender-cli and vice-versa, and I would argue that the `mender-cli` recording is a separate feature from what is provided by the backend. Thoughts?

File: 09.Add-ons/01.Remote-Terminal/docs.md (hunk 2 above)
Comment (alfrunes) [reply 09.Add-ons/01.Remote-Terminal/docs.md:81]: > Also technically speaking it is not the auditlogs that contain the session recording as it is.

I think the technical details does not matter in this context. The session log is so tightly coupled to audit log in the UI that I think we should treat it like so in the documentation.

**Comment by alfrunes** [comment]
@merlin-northern I tried to address most of your comments.
I also did some re-phrasing -- changing "Audit logs" to singular "Audit log".

File: 09.Add-ons/01.Remote-Terminal/40.Security/docs.md (hunk 3 above)
Comment (merlin-northern) [reply 09.Add-ons/01.Remote-Terminal/40.Security/docs.md]: mender-cli: I meant to mention that the sessions can also be recorded in the CLI :) "The same holds" referred to the session logging in general. sorry for the confusion.

Download from the UI: Manuel worked so hard on the downloader, so I would at least mention that in the UI you can download the recording. And good point to mention that the recordings are not compatible, they are the one from the UI is a html5 recording and the one from the CLI is terminal output zipped with custom metadata header.

File: 09.Add-ons/01.Remote-Terminal/40.Security/docs.md [hunk 5]
```diff
@@ -50,6 +50,8 @@ Remote Terminal does not open nor listen on any port: the device initiates the c
 !!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
 !!! for an overview of all Mender plans and features.
 
-Audit logs hold a list of every start and stop of a session. You can access this information
-in the Mender UI.
-
+Every remote terminal invocation is saved in the audit logs. The logs
+include the session- start and stop events as well as the terminal output. The
```
Comment (merlin-northern) [inline 09.Add-ons/01.Remote-Terminal/40.Security/docs.md:54]: is '-' on purpose in "session- "? if so then why there is space after it? (feels a bit confusing as it is now)

**Review by merlin-northern — APPROVED** [review]
lgtm, I leave the mention of CLI recording support at your discretion as well as the UI recording download.

File: 09.Add-ons/01.Remote-Terminal/40.Security/docs.md (hunk 5 above)
Comment (alfrunes) [reply 09.Add-ons/01.Remote-Terminal/40.Security/docs.md:54]: No, this hyphen does not belong here.

**Pull request merged**